### ISSUE TITLE
🚧Create service to get PR branch info

### ISSUE THREAD

**Issue by CalvinWilkinson** [KinsonDigital/CICD#22]
### Complete The Item Below

- [X] I have updated the title without removing the 🚧 emoji.

### Description

**Problem:**
Currently, getting pull request branch information will only work by using the `GitHubActions.HeadRef` or `GitHubActions.BaseRef`.  Using this is only good in the GitHub environment because that is the only time the `GitHubActions` instance is _**NOT NULL**_.
This means that running pull request related stuff locally, it becomes and issue because the object instance is null.

**Solution:**
To allow this functionality to work, create a service that will detect if the build is a server build or local build.  If it is a server build, use the `GitHubActions` instance, if it is a local build, use the `GitHubClient` to interact with the API to be able get the branch information required.

To maximize testability of the new service, create a thin wrapper around the `GitHubActions` object that can be injected into the new service for us.  This way we are not found to server only state and can easily maximize our testability.

Once the new service is complete, refactor the use cases throughout the code base to use the new service instead of the `GitHubActions` object.

### Acceptance Criteria

- [ ] Thin wrapper service around `GitHubActions` object created
- [ ] New service created
  - [ ] Service named `PullRequestService`
  - [ ] Source branch property created to represent **HeadRef**
  - [ ] Target branch property created to represent **TargetBranc**
  - [ ] Service unit tests created and passing
  - [ ] `GitHubActions` branch use cases replaced
  - [ ] Dependency injection setup

### ToDo Items

- [X] Change type labels added to this issue.  Refer to the _**Change Type Labels**_ section below.
- [X] Priority label added to this issue.  Refer to the _**Priority Type Labels**_ section below.
- [X] Issue linked to the correct project _(if applicable)_.
- [ ] Issue linked to the correct milestone _(if applicable)_.
- [ ] Draft pull request created and linked to this issue _(only required with code changes)_.

### Additional Information:

**_<details closed><summary>Change Type Labels</summary>_**

| Change Type         | Label                |
|---------------------|----------------------|
| Bug Fixes           | `🐛bug` |
| Breaking Changes    | `🧨breaking changes` |
| New Feature         | `✨new feature` |
| Workflow Changes    | `workflow` |
| Code Doc Changes    | `🗒️documentation/code` |
| Product Doc Changes | `📝documentation/product` |
</details>

**_<details closed><summary>Priority Type Labels</summary>_**

| Priority Type       | Label             |
|---------------------|-------------------|
| Low Priority        | `low priority`    |
| Medium Priority     | `medium priority` |
| High Priority       | `high priority`   |
</details>

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct.
